staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -174,14 +175,43 @@ func (a customResourceStrategy) Validate(ctx context.Context, obj runtime.Object
 
 		// validate x-kubernetes-validations rules
 		if celValidator, ok := a.celValidators[v]; ok {
-			err, _ := celValidator.Validate(ctx, nil, a.structuralSchemas[v], u.Object, cel.RuntimeCELCostBudget)
-			errs = append(errs, err...)
+			// if contains OpenAPIv3 type/maxLength/maxItems/required/wrong type field validation failures, skip cel validation
+			if hasBlockingErr(errs) {
+				errs = append(errs, field.Invalid(nil, nil, "Blocking error occurred above, skip CEL validation"))
+			} else {
+				err, _ := celValidator.Validate(ctx, nil, a.structuralSchemas[v], u.Object, cel.RuntimeCELCostBudget)
+				errs = append(errs, err...)
+			}
 		}
 	}
 
 	return errs
 }
 
+func hasBlockingErr(errs field.ErrorList) bool {
```

Comment:
Add a `underFieldPath string` and then do a `if strings.HasPrefix(err.Path, underFieldPath)` to decide which errors are relevant to a particular set of validation rules?

Reply:
Current we skip all cel validation as long as an type/maxLength/maxItems/MaxProperties/required error occurred. With the blocking error checking added in strategy, I am not sure how to check field based on validation rule fieldpath(And adding checking for filedPath of cel validation rules requires to construct celValidator which will add extra step).

Reply:
I have updated with fieldPath check